web/elm/src/Views/TopBar.elm

```diff
@@ -194,17 +208,56 @@ jobBreadcrumb jobName =
         )
 
 
-resourceBreadcrumb : String -> Html Message
-resourceBreadcrumb resourceName =
-    Html.li
-        (id "breadcrumb-resource" :: Styles.breadcrumbItem False)
+resourceBreadcrumb : Concourse.ResourceIdentifier -> Html Message
+resourceBreadcrumb resource =
+    Html.a
+        ([ id "breadcrumb-resource"
+         , href <|
+            Routes.toString <|
+                Routes.resourceRoute resource Nothing
+         ]
+            ++ Styles.breadcrumbItem True
+        )
         (breadcrumbComponent
             { icon =
                 { component = Assets.ResourceComponent
                 , widthPx = 32
                 , heightPx = 17
                 }
-            , name = resourceName
+            , name = resource.resourceName
+            }
+        )
+
+
+causalityBreadCrumb : Concourse.VersionedResourceIdentifier -> Concourse.CausalityDirection -> Concourse.Version -> Html Message
+causalityBreadCrumb rv direction version =
+    let
+        component =
+            case direction of
+                Concourse.Downstream ->
+                    Assets.DownstreamCausalityComponent
+
+                Concourse.Upstream ->
+                    Assets.UpstreamCausalityComponent
+
+        name =
+            String.join "," <| Concourse.versionQuery version
```

Comment:
Wonder if we should do a similar truncation for resource version components here - otherwise, the breadcrumb can be really long and look a bit wonky
<img width="1904" alt="Screen Shot 2021-06-11 at 4 01 50 PM" src="https://user-images.githubusercontent.com/26583442/121742937-8ba74100-cace-11eb-872a-af5b0849ca60.png">

Reply:
I feel like any non-css based solutions here would have trouble based on the size of the users device. We were able to use it in the SVG because the user can pan around, and we're only concerned with the minimum number of character to uniquely identify a resource.

Tried to solve this in https://github.com/concourse/concourse/pull/7125/commits/6786808e5d03fa53b2f8df8bbc41e0f7daf54a7f, but that only solves the problem if the very last breadcrumb overflows. Which should be fine if we assume pipeline and resource names to be somewhat short, and the only long breadcrumbs are the resource versions

Reply:
Maybe we could take the approach that we tried to take initially with the resource versions and wrap each key-value pair in a `span` that has a `max-width` and applies ellipsis to text if it overflows. So, the version breadcrumb may end up looking like:

```
digest:sha256:abc123foobar..., tag:branch-resource-causal...
```

Not a perfect solution as it can still wrap to multiple lines, though.